internal/service/datazone/environment_profile.go

```diff
@@ -0,0 +1,361 @@
+		state.AwsAccountRegion = flex.StringToFramework(ctx, out.AwsAccountRegion)
+		state.CreatedAt = flex.TimeToFramework(ctx, out.CreatedAt)
+		state.CreatedBy = flex.StringToFramework(ctx, out.CreatedBy)
+		state.Description = flex.StringToFramework(ctx, out.Description)
+		state.DomainIdentifier = flex.StringToFramework(ctx, out.DomainId)
+		state.EnvironmentBlueprintIdentifier = flex.StringToFramework(ctx, out.EnvironmentBlueprintId)
+		state.Id = flex.StringToFramework(ctx, out.Id)
+		state.Name = flex.StringToFramework(ctx, out.Name)
+		state.ProjectIdentifier = flex.StringToFramework(ctx, out.ProjectId)
+		state.UpdatedAt = flex.TimeToFramework(ctx, out.UpdatedAt)
+
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceEnvironmentProfile) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	conn := r.Meta().DataZoneClient(ctx)
+	var state dEnvProfile
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	_, err := conn.DeleteEnvironmentProfile(ctx, &datazone.DeleteEnvironmentProfileInput{
+		DomainIdentifier: state.DomainIdentifier.ValueStringPointer(),
+		Identifier:       state.Id.ValueStringPointer(),
+	})
+
+	if err != nil && !errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
+			err.Error(),
+		)
+		return
+	}
+}
+
+func (r *resourceEnvironmentProfile) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, ":")
+
+	if len(parts) != 4 {
+		resp.Diagnostics.AddError("Resource Import Invalid ID", fmt.Sprintf(`Unexpected format for import ID (%s), use: "DomainIdentifier:Id"`, req.ID))
+	}
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("domain_identifier"), parts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(names.AttrID), parts[1])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("environment_blueprint_identifier"), parts[2])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_identifier"), parts[3])...)
+}
+
+func findEnvironmentProfileByID(ctx context.Context, conn *datazone.Client, id string, domain_id string) (*datazone.GetEnvironmentProfileOutput, error) {
+	in := &datazone.GetEnvironmentProfileInput{
+		Identifier:       aws.String(id),
+		DomainIdentifier: aws.String(domain_id),
+	}
+
+	out, err := conn.GetEnvironmentProfile(ctx, in)
+	if err != nil {
+		if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+			return nil, &retry.NotFoundError{
+				LastError:   err,
+				LastRequest: in,
+			}
+		}
+
+		return nil, err
+	}
+
+	if out == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+type dEnvProfile struct {
```

Comment:
The typical naming convention for this struct is `<resourceName>Data` (ex. `environmentProfileData`)